sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -852,6 +852,24 @@ class DataFrame private[sql](
   }
 
   /**
+   * Returns a new [[DataFrame]] with a column dropped.
```

Comment:
document that this is no-op if the column doesn't exist. Can you also add this line to withColumnRenamed?